src/app/(main)/note/[id]/page.tsx

```diff
@@ -1,7 +1,47 @@
+import PostHeader from "@/components/shared/Post/PostHeader";
 import React from "react";
+import { Separator } from "@/components/ui/separator";
 
-const Note = () => {
-  return <div>Note</div>;
+import { getPost } from "~/lib/actions/posts";
+import { getPostDetails } from "~/lib/helpers/postDetails";
+import Markdown from "react-markdown";
+import CodeBlock from "@/components/shared/Post/CodeBlock/CodeBlock";
+import Link from "next/link";
+import Image from "next/image";
+import ResourcesAndLinks from "@/components/shared/Post/ResourcesAndLinks";
+import KeyTakeawaysPreview from "@/components/shared/Post/KeyTakeaways";
+import StepsToFollowPreview from "@/components/shared/Post/StepsToFollowPreview";
+
+const Note = async ({ params }: { params: { id: string } }) => {
+  const noteId = Number(params.id);
+  const noteFromServer = await getPost(noteId);
+  console.log(noteFromServer);
+  const { icon, backgroundColor, textColor } = getPostDetails(noteFromServer);
+
+  return (
+    <section className="flex min-h-screen w-6/12 flex-col justify-start gap-2 p-4">
```

Comment:
```suggestion
    <section className="flex min-h-screen w-6/12 flex-col justify-start gap-5 p-4">
```
Let's play around with some slightly larger gaps here. I think it looks a lot nicer and closer to the design:

## With larger gap:
![image](https://github.com/dalbanhi/git-note/assets/4108484/6faadd27-a452-43b0-9207-a57eab1463a8)


## Without larger gap:
![image](https://github.com/dalbanhi/git-note/assets/4108484/658eb713-37b5-442e-ad5d-82adf742c911)